osu.Game/Rulesets/Replays/FramedReplayInputHandler.cs

```diff
@@ -123,9 +123,7 @@ private bool inImportantSection
             // if we have a next frame, check if it is before or at the current time in playback, and advance time to it if so.
             if (next != null)
             {
-                int compare = time.CompareTo(next.Time);
-
-                if (compare == 0 || compare == currentDirection)
+                if (currentDirection == 1 ? next.Time <= time : time <= next.Time)
```

Comment:
Honestly I'd prefer to revert this and stick `Math.Sign` on the comparison result if anything. The way this is written makes my head hurt worse than the original (which I've read a few times already and it's already quite a headache).